### ISSUE TITLE
Code Fix for NUnit2010 on Ref Structs Creates CS0306

### ISSUE THREAD

**Issue by manfred-brands** [nunit/nunit.analyzers#549]
Relates to #475 where the detected code was `actual == expected`.
However NUnit2010 is also raised for `actual.Equals(expected)`
This was addressed partly in #477 but only if the argument was a `ref readonly struct` not if the instance was one.

```csharp
[Test]
public void TestMethod()
{
    ReadOnlySpan<char> actual = GetPart();
    Assert.That(actual.Equals(""Text""), Is.True);
}

ReadOnlySpan<char> GetPart() => ""Some Text"".AsSpan(4);
```

Here the method `ReadOnlySpan<char>.Equals(string)` is called.

The CodeFix converts this into: `Assert.That(actual, Is.EqualTo(expected));`
which raises: ` error CS0306: The type 'ReadOnlySpan<char>' may not be used as a type argument`
